exporters/otlp/test/otlp_log_recordable_test.cc

```diff
@@ -0,0 +1,237 @@
+  rec.SetBody(name);
+  EXPECT_EQ(rec.log_record().body().string_value(), name);
+}
+
+TEST(OtlpLogRecordable, SetTraceId)
+{
+  OtlpLogRecordable rec;
+  uint8_t trace_id_bin[opentelemetry::trace::TraceId::kSize] = {
+      '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'};
+  std::string expected_bytes;
+  expected_bytes.assign(
+      reinterpret_cast<char *>(trace_id_bin),
+      reinterpret_cast<char *>(trace_id_bin) + opentelemetry::trace::TraceId::kSize);
+  rec.SetTraceId(opentelemetry::trace::TraceId{trace_id_bin});
+  EXPECT_EQ(rec.log_record().trace_id(), expected_bytes);
+}
+
+TEST(OtlpLogRecordable, SetSpanId)
+{
+  OtlpLogRecordable rec;
+  uint8_t span_id_bin[opentelemetry::trace::SpanId::kSize] = {'7', '6', '5', '4',
+                                                              '3', '2', '1', '0'};
+  std::string expected_bytes;
+  expected_bytes.assign(
+      reinterpret_cast<char *>(span_id_bin),
+      reinterpret_cast<char *>(span_id_bin) + opentelemetry::trace::SpanId::kSize);
+  rec.SetSpanId(opentelemetry::trace::SpanId{span_id_bin});
+  EXPECT_EQ(rec.log_record().span_id(), expected_bytes);
+}
+
+TEST(OtlpLogRecordable, SetResource)
+{
+  OtlpLogRecordable rec;
+  const std::string service_name_key = "service.name";
+  std::string service_name           = "test-otlp";
+  rec.SetResource(service_name_key, service_name);
+
+  auto proto_resource     = rec.ProtoResource();
+  bool found_service_name = false;
+  for (int i = 0; i < proto_resource.attributes_size(); i++)
+  {
+    auto attr = proto_resource.attributes(static_cast<int>(i));
+    if (attr.key() == service_name_key && attr.value().string_value() == service_name)
+    {
+      found_service_name = true;
+    }
+  }
+  EXPECT_TRUE(found_service_name);
+}
+
+TEST(OtlpLogRecordable, DefaultResource)
+{
+  OtlpLogRecordable rec;
+
+  auto proto_resource      = rec.ProtoResource();
+  int found_resource_count = 0;
+  for (int i = 0; i < proto_resource.attributes_size(); i++)
+  {
+    auto attr = proto_resource.attributes(static_cast<int>(i));
+    if (attr.key() ==
+        opentelemetry::sdk::resource::attr(OTEL_CPP_CONST_HASHCODE(AttrTelemetrySdkLanguage)))
+    {
+      EXPECT_EQ(attr.value().string_value(), "cpp");
+      ++found_resource_count;
+    }
+    else if (attr.key() ==
+             opentelemetry::sdk::resource::attr(OTEL_CPP_CONST_HASHCODE(AttrTelemetrySdkName)))
+    {
+      EXPECT_EQ(attr.value().string_value(), "opentelemetry");
+      ++found_resource_count;
+    }
+    else if (attr.key() ==
+             opentelemetry::sdk::resource::attr(OTEL_CPP_CONST_HASHCODE(AttrTelemetrySdkVersion)))
+    {
+      EXPECT_EQ(attr.value().string_value(), OPENTELEMETRY_SDK_VERSION);
+      ++found_resource_count;
+    }
+  }
+  EXPECT_EQ(3, found_resource_count);
+}
+
+// Test non-int single types. Int single types are tested using templates (see IntAttributeTest)
+TEST(OtlpLogRecordable, SetSingleAtrribute)
```

Comment:
```suggestion
TEST(OtlpLogRecordable, SetSingleAttribute)
```

Reply:
Done. I also fix the spell error in `otlp_recordable_test.cc`